app/controllers/curation_concerns/single_use_links_viewer_controller.rb

```diff
@@ -0,0 +1,21 @@
+module CurationConcerns
+  class SingleUseLinksViewerController < ApplicationController
+    include CurationConcerns::SingleUseLinksViewerControllerBehavior
+
+    class Ability
+      include CanCan::Ability
+
+      attr_reader :single_use_link
+
+      def initialize(user, single_use_link)
+        @user = user || ::User.new
+
+        @single_use_link = single_use_link
+
+        can :read, ActiveFedora::Base do |obj|
+          single_use_link.valid? && single_use_link.itemId == obj.id && single_use_link.destroy!
+        end if single_use_link
```

Comment:
@grosscol I do not think it is a mistake.  It means only run the block if single_use_link is defined.